src/Nethermind/Nethermind.Optimism/CL/P2P/OptimismCLP2P.cs

```diff
@@ -175,26 +272,61 @@ private bool TryValidateAndDecodePayload(byte[] msg, [MaybeNullWhen(false)] out
         return true;
     }
 
+    private async Task<ISession?> TryDialPeer(Multiaddress remotePeer, CancellationToken token)
+    {
+        try
+        {
+            CancellationTokenSource cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(token);
+            cancellationTokenSource.CancelAfter(TimeSpan.FromSeconds(5));
+            return await _localPeer!.DialAsync(remotePeer, cancellationTokenSource.Token);
+        }
+        catch (Exception e)
+        {
+            if (_logger.IsTrace) _logger.Trace($"Unable to Dial peer {remotePeer}. Exception: {e.Message}");
+            return null;
+        }
+    }
+
     public async Task Run(CancellationToken token)
     {
         if (_logger.IsInfo) _logger.Info("Starting Optimism CL P2P");
 
         IPeerFactory peerFactory = _serviceProvider.GetService<IPeerFactory>()!;
-        string address = $"/ip4/{_config.P2PHost}/tcp/{_config.P2PPort}";
+        string address = $"/ip4/{_config.ClP2PHost}/tcp/{_config.ClP2PPort}";
```

Comment:
Probably need to detect public address or at least use 0.0.0.0 (127.0.0.1 not the coolest option). In contrast to json rpc endpoints, libp2p peer usually listens to public IP.

Reply:
Done